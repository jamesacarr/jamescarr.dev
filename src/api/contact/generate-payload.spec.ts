import generatePayload from './generate-payload';

describe('generatePayload', () => {
  it('generates the payload correctly', () => {
    const data = {
      name: 'John Doe',
      email: 'test@test.com',
      message: 'testing',
    };

    expect(generatePayload(data)).toMatchSnapshot();
  });
});
