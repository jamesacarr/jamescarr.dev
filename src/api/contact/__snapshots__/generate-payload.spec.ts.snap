// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generatePayload generates the payload correctly 1`] = `
Object {
  "attachments": Array [
    Object {
      "blocks": Array [
        Object {
          "fields": Array [
            Object {
              "text": "*Name:*
John Doe",
              "type": "mrkdwn",
            },
            Object {
              "text": "*Email:*
<mailto:test@test.com|test@test.com>",
              "type": "mrkdwn",
            },
          ],
          "type": "section",
        },
        Object {
          "text": Object {
            "text": "*Message:*
testing",
            "type": "mrkdwn",
          },
          "type": "section",
        },
      ],
    },
  ],
  "text": "You have a new message from the contact form:",
}
`;
